public/app/features/alerting/unified/state/actions.ts

```diff
@@ -1,25 +1,65 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
 import { ThunkResult } from 'app/types';
-import { RuleNamespace } from 'app/types/unified-alerting';
+import { RuleLocation, RuleNamespace } from 'app/types/unified-alerting';
+import { RulerRulesConfigDTO } from 'app/types/unified-alerting-dto';
 import { fetchRules } from '../api/prometheus';
-import { getRulesDataSources, GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
+import { deleteRulerRulesGroup, fetchRulerRules, fetchRulerRulesNamespace, setRulerRuleGroup } from '../api/ruler';
+import { getAllRulesSourceNames, isCloudRulesSource } from '../utils/datasource';
 import { withSerializedError } from '../utils/redux';
+import { hashRulerRule } from '../utils/rules';
 
-/*
- * Will need to be updated to:
- *
- * 1. Fetch grafana managed rules when the endpoint becomes available
- * 2. Reconcile with rules from the ruler where ruler is available
- */
-
-export const fetchRulesAction = createAsyncThunk(
-  'unifiedalerting/fetchRules',
+export const fetchPromRulesAction = createAsyncThunk(
+  'unifiedalerting/fetchPromRules',
   (rulesSourceName: string): Promise<RuleNamespace[]> => withSerializedError(fetchRules(rulesSourceName))
 );
 
-export const fetchRulesFromAllSourcesAction = (): ThunkResult<void> => {
+export const fetchRulerRulesAction = createAsyncThunk(
+  'unifiedalerting/fetchRulerRules',
+  (rulesSourceName: string): Promise<RulerRulesConfigDTO | null> => {
+    return withSerializedError(fetchRulerRules(rulesSourceName));
+  }
+);
+
+export function fetchAllPromAndRulerRules(force = false): ThunkResult<void> {
+  return (dispatch, getStore) => {
+    const { promRules, rulerRules } = getStore().unifiedAlerting;
+    getAllRulesSourceNames().map((name) => {
+      if (force || !promRules[name]?.loading) {
+        dispatch(fetchPromRulesAction(name));
+      }
+      if (force || !rulerRules[name]?.loading) {
+        dispatch(fetchRulerRulesAction(name));
+      }
+    });
+  };
+}
+
+export function deleteRuleAction(ruleLocation: RuleLocation): ThunkResult<void> {
+  /*
+   * fetch the rules group from backend, delete group if it is found and+
+   * reload ruler rules
+   */
```

Comment:
Why do the rules need to be fetched from the backend again?

Reply:
deleting is done by re-uploading the entire group with the deleted rule removed. So we fetch the latest version to make sure that we don't accidentally upload an older version in case something has changed since last fetch